projects/arc-lib/README.md

```diff
@@ -0,0 +1,138 @@
+  API endpoint for triggering forget password functionality,`retrieving` current user's data,authentication token,initiating a Google Auth login flow and logout etc.
+
+- Guards: Two guards provided are `LoggedInGuard` and `AuthGuard` used to control
+  access to certain routes or pages in the application.Guards can be used to check if the user has permission to access a certain route, if the user is logged in, or to perform other types of authentication or authorization check.
+
+- Models: We are using 4 models named `logged-in-user.model`,`loggedin-response.model`,
+  `login. model` and `token-response.model` that represents user's data when they are logged in,response from a login API call(code),data that is required for a user to authenticate or log
+  in to a system and the the response from server after a user gets authenticated
+  and obtains a new access token and refresh token.
+
+## Decorator:
+
+We are using 2 decorators named `required.decorator` and `validate.decorator`.
+
+- The `required` decorator is used to mark properties in a model as required and adds the
+  property name to a list of required properties stored in metadata.
+- The `validate` decorator is used as a class decorator to validate the property constraints of
+  a parameterized constructor. The required decorator adds metadata to the target constructor and validate uses `Reflect.getMetadata` to retrieve the required properties and validate them in the constructor.
+  These decorators help to ensure that required properties are not left undefined or null.
+
+## Interceptors:
+
+The boilerplate using 3 Interceptors named `auth.interceptor`,`error.interceptor` and `session-recovery.interceptor`. These interceptors are responsible for adding an authorization header to requests that require authentication, checks for errors that occur during HTTP requests and
+displays error messages and also refreshes the user's authentication token when the session gets expired.
+
+## Localization:
+
+The boilerplate using i18n module as a language translator where The boilerplate using 3 files as follows :
+
+- Enums: `language.enum` defines the set of named values for various languages.
+
+- Module: The boilerplate using `localization.module` while using the module The boilerplate using data from enum too and according to enums value we translate the language and `translation service`
+
+- Service: The translationService provides localization functionality to the application. It
+  depends on `@ngx-translate/core` library to handle translations allowing users to select their preferred language for the application & to provide appropriate translations based on that preference.
+
+## Env Resolver Service:
+
+The purpose of this service (implements the Resolve interface and is responsible for resolving the environment configuration ) is to retrieve environment configuration data from a store (`SystemStoreFacadeService`) and make it available to components before they are displayed.
```

Comment:
havent we removed this or is it still there and being used?

Reply:
we are using this